src/main/java/au/org/ala/biocache/service/DownloadService.java

```diff
@@ -1266,53 +1284,49 @@ public DownloadDetailsDTO call() throws Exception {
 
 
                                 String archiveFileLocation = biocacheDownloadUrl + File.separator + URLEncoder.encode(currentDownload.getFileLocation().replace(biocacheDownloadDir + "/",""), "UTF-8").replace("%2F", "/").replace("+", "%20");
-
+                                final String searchUrl = generateSearchUrl(currentDownload.getRequestParams());
                                 String doiStr = "";
-                                String emailBody;
                                 String emailTemplate;
-                                String downloadFileLocation;
-                                String officialFileLocation = "";
+                                String emailTemplateFile;
+                                Map<String, String> substitutions = new HashMap<>();
+                                substitutions.put(START_DATE_TIME, currentDownload.getStartDateString());
+                                substitutions.put(QUERY_TITLE, currentDownload.getRequestParams().getDisplayString());
+                                substitutions.put(SEARCH_URL, searchUrl);
+                                substitutions.put(DOI_FAILURE_MESSAGE,  biocacheDownloadDoiFailureMessage);
+
                                 if(mintDoi && doiResponseList != null && !doiResponseList.isEmpty() && doiResponseList.get(0) != null) {
 
                                     CreateDoiResponse doiResponse;
                                     doiResponse = doiResponseList.get(0);
                                     try {
                                         doiService.updateFile(doiResponse.getUuid(), currentDownload.getFileLocation());
                                         doiStr = doiResponse.getDoi();
-                                        emailTemplate = biocacheDownloadDoiEmailTemplate;
+                                        if (currentDownload.getRequestParams().getEmailTemplate() == DEFAULT_SELECTOR)
```

Comment:
Use braces for this if condition to match the rest of the biocache-service style.